pkg/activator/net/throttler_test.go

```diff
@@ -449,11 +456,15 @@ func TestThrottlerSuccesses(t *testing.T) {
 			tryContext, cancel2 := context.WithTimeout(context.Background(), 1*time.Second)
 			defer cancel2()
 
+			waitGrp := sync.WaitGroup{}
+			waitGrp.Add(tc.requests)
 			resultChan := throttler.try(tryContext, tc.requests, func(string) error {
-				// Simulate proxying.
-				time.Sleep(50 * time.Millisecond)
+				waitGrp.Done()
+				// Wait for all requests to reach the Try function before proceeding.
+				waitGrp.Wait()
```

Comment:
little sad that if we mess the throttler/a test case up here we'll deadlock the tests rather than erroring.. e.g. if you accidentally set requests to `2` instead of `1` in the "single healthy clusterIP" case (because it's CC=1) the test deadlocks here instead of failing nicely, even though the context has a timeout. Maybe we could add an explicit timeout here?

Reply:
Well, we can probably complicate things more by replacing wait group with err group (WithContext kind) and wait on it.

Reply:
I'm mostly concerned about fixing the race conditions of the test here. Not sure it matters a ton here.

Reply:
I mean it's certainly not the end of the world either way, but seems like we could replace the `WithTimeout` a few lines up that's - I think - now redundant with a `select { case <- time.After(1 * time.Second): t.Fatal() }` and have an actual timeout instead of a deadlock.